test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs

```diff
@@ -44,7 +44,25 @@ public async Task GivenANewSearchParam_WhenReindexingComplete_ThenResourcesSearc
             await Client.CreateAsync(Samples.GetJsonSample<Patient>("Patient"));
 
             // POST a new Search parameter
-            await Client.CreateAsync(searchParam);
+            FhirResponse<SearchParameter> searchParamPosted = null;
+            try
+            {
+                searchParamPosted = await Client.CreateAsync(searchParam);
+            }
+            catch (Exception ex)
+            {
+                // if the SearchParameter exists, we should delete it and recreate it
+                var searchParamBundle = await Client.SearchAsync(ResourceType.SearchParameter, $"url={searchParam.Url}");
+                if (searchParamBundle.Resource?.Entry[0] != null && searchParamBundle.Resource?.Entry[0].Resource.ResourceType == ResourceType.SearchParameter)
+                {
+                    await DeleteSearchParameterAndVerify(searchParamBundle.Resource?.Entry[0].Resource as SearchParameter);
+                    searchParamPosted = await Client.CreateAsync(searchParam);
+                }
+                else
+                {
+                    throw ex;
```

Comment:
>throw ex [](start = 20, length = 8)

nit: I think this is antipattern.
`ex` is not used anywhere and just got rethrown which would lead to lost of original stack. Not that important in tests, but i think it's nice thing to avoid overall.

here some details:
[https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2200](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2200)